src/components/Detail/Detail.tsx

```diff
@@ -0,0 +1,81 @@
+import { useRouter } from 'next/router';
+import classnames from 'classnames';
+
+import { Row } from '../Row/Row';
+import { trpc } from '../../utils/trpc';
+import style from './Detail.module.scss';
+
+export const DetailOfActivity = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const utils = trpc.useContext();
+
+  const activityData = trpc.activities.read.useQuery({ id: Number(id) }).data;
```

Comment:
Learn this pattern; it could be useful, especially for trpc/react-query, as this is very common.

```suggestion
  const { data: activityData } = trpc.activities.read.useQuery({ id: Number(id) });
```